server/pulp/plugins/distributor.py

```
@@ -187,6 +191,20 @@ def create_consumer_payload(self, repo, config, binding_config):
         """
         return {}
 
+    @staticmethod
+    def ensure_all_units_downloaded(repo_id):
+        """
+        Checks the database to make sure all units in the repo have been downloaded. If not, raises
+        an exception.
+
+        :param repo_id: repository id
+        :type  repo_id: basestring
+
+        :raises exceptions.PulpCodedException:  if any unit in the repo has un-downloaded files
+        """
+        if not repo_controller.has_all_units_downloaded(repo_id):
+            raise exceptions.PulpCodedException(error_code=error_codes.PLP0045)
```

I like how this raises an Exception rather than using a return code. +1